Public/Src/Utilities/UnitTests/Utilities/AbsolutePathTestsUnix.cs

```diff
@@ -34,8 +34,13 @@ public void TryCreate()
             XAssert.IsTrue(AbsolutePath.TryCreate(pt, @"/usr/local", out p));
             XAssert.AreEqual(@"/usr/local", p.ToString(pt));
 
-            XAssert.IsFalse(AbsolutePath.TryCreate(pt, @"/..", out p));
-            XAssert.IsFalse(AbsolutePath.TryCreate(pt, @"/../..", out p));
+
+            XAssert.IsTrue(AbsolutePath.TryCreate(pt, @"/..", out p));
+            XAssert.AreEqual(@"/", p.ToString(pt));
+            XAssert.IsTrue(AbsolutePath.TryCreate(pt, @"/../..", out p));
+            XAssert.AreEqual(@"/", p.ToString(pt));
+            XAssert.IsTrue(AbsolutePath.TryCreate(pt, @"/usr/etc/aaa/../../../../../../", out p));
+            XAssert.AreEqual(@"/", p.ToString(pt));

```

Comment:
Do we really want this behavior? 
I agree on the absolute path case since that matches the win32 api's... But I'm not 100% sure we want this for relative paths, it seems that going out of a relative path should remain an error...

Reply:
And I should have looked better :D This is unix paths, not relative paths...


---
In reply to: [298701845](https://github.com/microsoft/BuildXL/pull/533#discussion_r298701845) [](ancestors = 298701845)

Reply:
Indeed, so if the path is passed into the RelativePath's tryCreate directly, it will error out. But if it is coming through an AbsolutePath's tryCreate call, then we allow it to go through since we know that there is a root directory to fall back to.

Reply:
Thanks, sorry for not reading carefully enough :)